project_event/models/project_task.py

```diff
@@ -396,20 +396,26 @@ def create_children_from_activity_create(self, children, parent_id):
 
     @api.multi
     def write(self, vals):
-        if self.is_activity():
-            return self.write_activity(vals)
-        else:
-            self.update_reservation_event(vals)
-            return super(Task, self).write(vals)
+        super(Task, self).write(vals)
+        for rec in self:
+            if rec.is_activity():
+                return rec.write_activity(vals)
+            else:
+                rec.update_reservation_event(vals)
+                return super(Task, rec).write(vals)
 
     def is_activity(self):
-        return self.activity_task_type == 'activity'
+        self.ensure_one()
```

Comment:
if it is only one record why we need to use for rec in self?

Reply:
@rimbendhaou  Changed Done